engine/lib/phases/phase_functionalize_loops.ml

```diff
@@ -259,6 +312,39 @@ struct
                 (M.pat_PWild ~span ~typ:unit.typ, unit)
           in
           let body = dexpr body in
+          let { body; annotation = annotation1 } =
+            extract_loop_annotation body
+          in
+          let { body; annotation = annotation2 } =
+            extract_loop_annotation body
+          in
+          let invariant =
+            match (annotation1, annotation2) with
+            | Some (LoopInvariant { index_pat = None; invariant }), _
+            | _, Some (LoopInvariant { index_pat = None; invariant }) ->
+                invariant
+            | _ -> MS.expr_Literal ~typ:TBool (Bool true)
+          in
+          let variant =
+            match (annotation1, annotation2) with
+            | Some (LoopVariant variant), _ | _, Some (LoopVariant variant) ->
+                variant
+            | _ ->
+                let kind = { size = S32; signedness = Unsigned } in
+                let e =
+                  UB.M.expr_Literal ~typ:(TInt kind) ~span:body.span
+                    (Int { value = "0"; negative = false; kind })
+                in
+                UB.call Rust_primitives__hax__int__from_machine [ e ] e.span
+                  (TApp
+                     {
+                       ident =
+                         `Concrete
+                           (Concrete_ident.of_name ~value:false
+                              Hax_lib__int__Int);
+                       args = [];
+                     })
+          in
```

Comment:
Can you hoist this logic out of the function?
would be great to have something like a function that returns a tuple or a record of:
 - an option of an invariant
 - an option of a decrease clause

(and that fails if one provides two same clause)